pkg/controllers/cspi-controller/handler.go

```diff
@@ -241,6 +227,51 @@ func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance
 	return nil
 }
 
+// updateROMode sets/unsets the pool readonly mode property. It does the following changes
+// 1. If pool used space reached to roThresholdLimit then pool will be set to readonly mode
+// 2. If pool was in readonly mode if roThresholdLimit/pool expansion was happened then it
+//    unsets the ReadOnly Mode.
+// NOTE: This function must be invoked after having the updated cspiStatus information
+func (c *CStorPoolInstanceController) updateROMode(
+	cspiStatus *cstor.CStorPoolInstanceStatus, cspi cstor.CStorPoolInstance) {
+	roThresholdLimit := cspi.Spec.PoolConfig.ROThresholdLimit
+	totalInBytes := cspiStatus.Capacity.Total.Value()
+	usedInBytes := cspiStatus.Capacity.Used.Value()
+	pool := zpool.PoolName()
+
+	usedPercentage := (usedInBytes * 100) / totalInBytes
+	if (int(usedPercentage) >= roThresholdLimit) &&
+		(roThresholdLimit != 0 && roThresholdLimit != 100) {
+		if !cspiStatus.ReadOnly {
```

Comment:
What is status is ReadOnly but pool is not readOnly ? I mean this could be a bad state but if we get into this state -- we won't be able to recover. 
Ans these things can happen due to API server unreachability.
Pool if needs to go into RO should go into RO even if the status is stale...
The idea is finally things should converge towards a good guy.

Reply:
We will not be dealing with stale/outdated info since we are getting info from the pool by doing zfs/zpool commands before calling this function
https://github.com/openebs/cstor-operators/blob/e1d33c5449ff0133b09441463c111428a67d02ae/pkg/controllers/cspi-controller/handler.go#L206